blocks/library/tag-cloud/block.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+import { stringify } from 'querystringify';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import {
+	Placeholder,
+	ToggleControl,
+	SelectControl,
+	withAPIData,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import TermList from './term-list';
+import InspectorControls from '../../inspector-controls';
+import BlockControls from '../../block-controls';
+import BlockAlignmentToolbar from '../../block-alignment-toolbar';
+
+class TagCloudBlock extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.setTaxonomy = this.setTaxonomy.bind( this );
+		this.toggleShowTagCounts = this.toggleShowTagCounts.bind( this );
+	}
+
+	getTaxonomies() {
+		const taxonomies = this.props.taxonomies.data;
+
+		if ( ! taxonomies ) {
+			return [];
+		}
+
+		return map( taxonomies, taxonomy => {
+			return {
+				value: taxonomy.slug,
+				label: taxonomy.name,
+			};
+		} );
+	}
+
+	setTaxonomy( taxonomySlug ) {
+		const { setAttributes } = this.props;
+		const taxonomies = this.props.taxonomies.data;
+		const taxonomy = taxonomies[ taxonomySlug ];
+
+		setAttributes( { taxonomySlug, taxonomyRestBase: taxonomy.rest_base } );
```

Comment:
I can't help but think that `taxonomyRestBase` is redundant with `taxonomySlug`.

You might be able to eliminate it in favor of just the slug, taking advantage of the `taxonomy` helper included in the argument passed to `withAPIData` callback:

https://github.com/WordPress/gutenberg/tree/master/components/higher-order/with-api-data#usage

Reply:
Thanks I wasn't aware of the `taxonomy` helper, it does simplify things a lot!